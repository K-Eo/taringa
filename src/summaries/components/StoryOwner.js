import React from 'react'
import { Link } from 'react-router-dom'

import './StoryOwner.css'

const StoryOwner = ({ children }) => {
  const handleOnClick = e => e.stopPropagation()

  return (
    <div className="StoryOwner">
      <span>Por </span>
      <Link to={`/u/${children}`} onClick={handleOnClick}>
        {children}
      </Link>
    </div>
  )
}

StoryOwner.defaultProps = {
  children: '#USER',
}

export default StoryOwner
