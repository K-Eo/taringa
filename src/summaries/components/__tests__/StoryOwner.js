import React from 'react'

import StoryOwner from '../StoryOwner'

describe('StoryOwner', () => {
  it('exists', () => {
    expect(StoryOwner).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<StoryOwner />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders children', () => {
    const wrapper = shallow(<StoryOwner>foo</StoryOwner>)

    expect(wrapper).toMatchSnapshot()
  })
})
