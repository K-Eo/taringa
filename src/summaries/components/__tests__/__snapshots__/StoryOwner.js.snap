// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryOwner renders children 1`] = `
<div
  className="StoryOwner"
>
  <span>
    Por 
  </span>
  <Link
    onClick={[Function]}
    replace={false}
    to="/u/foo"
  >
    foo
  </Link>
</div>
`;

exports[`StoryOwner renders default 1`] = `
<div
  className="StoryOwner"
>
  <span>
    Por 
  </span>
  <Link
    onClick={[Function]}
    replace={false}
    to="/u/#USER"
  >
    #USER
  </Link>
</div>
`;
